### PR TITLE
add popup player when below the fold in watch view

### PR DESCRIPTION
**Pull request by rigelk**
see #1345 for the initial PR.

corrections brought:
- no more page jump when activating the player
- responds to more events (especially for touch devices)
- rebased on develop
- no need to scroll back to the top to get the player back in place
- changed the setting string to be more descriptive
- make it disabled by default

still needed:
- expose its activation to non-authenticated users (will do in another PR)

cc @BO41

### CODE REVIEW

**Comment by BO41** [comment]
@rigelk nice! looks good :tada: 
**Thanks** for making an improved version :+1:

File: client/src/app/+my-account/my-account-settings/my-account-video-settings/my-account-video-settings.component.html [hunk 1]
```diff
@@ -29,5 +29,12 @@
     ></my-peertube-checkbox>
   </div>
 
+  <div class="form-group">
```
Comment (rigelk) [inline client/src/app/+my-account/my-account-settings/my-account-video-settings/my-account-video-settings.component.html]: This is an artifact from the merge which needs to be removed :x

File: client/src/app/videos/+video-watch/video-watch.component.ts [hunk 2]
```diff
@@ -139,7 +141,27 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
...
+      this.hotkeysService.add(this.hotkeys)
+
+      if (this.user.pipPlayer) {
+        eventStreams
+        .pipe(throttleTime(250))
+        .subscribe(() => this.onScroll())
+      }
+    } else {
+      eventStreams
+      .pipe(throttleTime(250))
```
[the remark this answers is not in the document]
Comment (rigelk) [reply client/src/app/videos/+video-watch/video-watch.component.ts]: it's already there, you are looking at a previous version of the code.

File: server/models/account/user.ts [hunk 3]
```diff
@@ -120,6 +121,12 @@ export class UserModel extends Model<UserModel> {
   @Column
   autoPlayVideo: boolean
 
+  @AllowNull(false)
+  @Default(true)
```
Comment (Chocobozzz) [inline server/models/account/user.ts]: `@Default(false)`?

File: client/src/app/videos/+video-watch/video-watch.component.ts [hunk 4]
```diff
@@ -534,4 +551,20 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
...
     }
   }
+
+  private onScroll () {
+    const playerElementWrapper = this.elementRef.nativeElement.querySelector('#video-element-wrapper')
+    // returns position of an element relative to the viewport
+    const position = playerElementWrapper.getBoundingClientRect()
+
+    // if the player is 100px from leaving screen, it pops out
+    if (position.bottom <= 50 && !this.playerPopedOut) {
```
Comment (Chocobozzz) [inline client/src/app/videos/+video-watch/video-watch.component.ts]: Maybe put `50` in a variable/constant

File: server/models/account/user.ts (hunk 3 above)
Comment (rigelk) [reply server/models/account/user.ts]: :+1:

**Comment by Chocobozzz** [comment]
Closing: we need a real PIP support (ie injecting the player inside a root component so that when we change the page, the players remains displayed).

**Pull request closed without merging**